.buildkite/pipeline.linux.yml

```diff
@@ -0,0 +1,139 @@
+steps:
+  - label: "build-gnu-x86"
+    commands:
+     - cargo build --release
+    retry:
+      automatic: false
+    agents:
+      platform: x86_64.metal
+    plugins:
+      - docker#v3.0.1:
+          image: "fandree/rust-vmm-dev"
+          always-pull: true
+
+  steps:
+  - label: "build-gnu-x86-mmap"
+    commands:
+     - cargo build --release --features=backend-mmap
+    retry:
+      automatic: false
+    agents:
+      platform: x86_64.metal
+    plugins:
+      - docker#v3.0.1:
+          image: "fandree/rust-vmm-dev"
+          always-pull: true
+
+  steps:
+  - label: "build-gnu-arm-mmap"
+    commands:
+     - cargo build --release --features=backend-mmap
+    retry:
+      automatic: false
+    agents:
+      platform: arm.metal
+    plugins:
+      - docker#v3.0.1:
+          image: "fandree/rust-vmm-dev"
+          always-pull: true
+
+  steps:
+  - label: "build-musl-arm-mmap"
+    commands:
+     - cargo build --release --features=backend-mmap --target aarch64-unknown-linux-musl
+    retry:
+      automatic: false
+    agents:
+      platform: arm.metal
+    plugins:
+      - docker#v3.0.1:
+          image: "fandree/rust-vmm-dev"
+          always-pull: true
+
+  - label: "style"
+    command: cargo fmt --all -- --check
+    retry:
+      automatic: false
+    agents:
+      platform: x86_64.metal
+    plugins:
+      - docker#v3.0.1:
+          image: "fandree/rust-vmm-dev"
+          always-pull: true
+
+  - label: "unittests-gnu-x86"
+    commands:
+     - cargo test
+    retry:
+      automatic: false
+    agents:
+      platform: x86_64.metal
+    plugins:
+      - docker#v3.0.1:
+          privileged: true
+          image: "fandree/rust-vmm-dev"
+          always-pull: true
+
+  - label: "unittests-gnu-x86-mmap"
```

Comment:
For unit tests we can do ```cargo test --all-features``` instead of having one step/feature. We can have `unittests-gnu-x86` and `unittests-gnu-x86-mmap` as one feature with the --all-features flag.

Reply:
Sure, works for me. This was one of the open questions I had before.

Reply:
Sorry, I didn't really get that. For unit tests I think it's perfectly fine to have it all features tested at once. I think only the builds should be separate.

Reply:
Great, thanks for clarifying this.